crates/polars-time/src/windows/group_by.rs

```diff
@@ -267,16 +270,24 @@ pub(crate) fn group_by_values_iter_lookbehind(
             // we have a complete lookbehind so we know that `i` is the upper bound.
             // Safety
             // we are in bounds
-            let slice = unsafe { time.get_unchecked_release(last_lookbehind_i..i) };
+            let slice = unsafe {
+                time.get_unchecked_release(last_lookbehind_i..i - n_duplicates - duplicates_count)
+            };
```

Comment:
this is the gist of the change - if `closed='left'`, then `slice` might not actually be partitioned by `!b.is_member(*v, closed_window)`. So, it's necessary to go back to the first element in the sequence of duplicates